### PR TITLE
feat: add apple pay support

### PR DESCRIPTION
**Pull request by douglaseggleton**
Add support for Apple Pay - also detects the version of the API being used and attempts to upgrade to version 5 (broadest set of card schemes), but at a minimum fallbacks to 3 (pre 3 has a subtly different api).

### CODE REVIEW

File: packages/embed/src/types.ts [hunk 1]
```diff
@@ -181,4 +181,19 @@ export type Message = { channel: string; data?: unknown } & (
...
     }
+  | {
+      type: 'appleAbortSession'
+    }
+  | {
+      type: 'appleCompletePayment'
+      data: boolean
+    }
+  | {
+      type: 'completeMerchantValidation'
```
Comment (gryevns) [inline packages/embed/src/types.ts:192]: Should this be name scoped to apple like the others? i.e. `appleCompleteMerchantValidation`

File: packages/embed/src/apple-pay/apple-pay.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    | 'completeMerchantValidation$'
+    | 'appleCompletePayment$'
+    | 'appleSessionError$'
+  >
+) => {
+  let session: ApplePaySession // only a single session at a time
+
+  subjectManager.startAppleSession$.subscribe((data) => {
+    try {
+      session = new ApplePaySession(3, data)
```
Comment (gryevns) [inline packages/embed/src/apple-pay/apple-pay.ts:25]: Does 3 represent the version? Does this need to be dynamic?

Comment (douglaseggleton) [reply packages/embed/src/apple-pay/apple-pay.ts:25]: 👀  Good spot!

File: packages/embed/src/types.ts (hunk 1 above)
Comment (douglaseggleton) [reply packages/embed/src/types.ts:192]: Yep - Ill make the udpate

**Comment by douglaseggleton** [comment]
@gryevns I've also renamed `startAppleSes..` to `appleStartSession` so they are all the same

**Review by gryevns — APPROVED** [review]

**Comment by gr4vy-code** [comment]
:rocket: PR was released in [`v1.5.0`](https://github.com/gr4vy/gr4vy-embed/releases/tag/v1.5.0) :rocket:

**Pull request merged**